lib/components/src/tooltip/WithTooltip.tsx

```diff
@@ -1,26 +1,38 @@
-import React from 'react';
-import PropTypes from 'prop-types';
+import React, { FunctionComponent, ReactNode } from 'react';
 import { styled } from '@storybook/theming';
 import { logger } from '@storybook/client-logger';
 import { withState, lifecycle } from 'recompose';
 import { document } from 'global';
 
 import TooltipTrigger from 'react-popper-tooltip';
 import Tooltip from './Tooltip';
+import { Placement } from 'popper.js';
 
 // A target that doesn't speak popper
-// prettier-ignore
-const TargetContainer = styled.div`
+const TargetContainer = styled.div<{ mode: string }>`
   display: inline-block;
-  cursor: ${props => props.mode === 'hover' ? 'default' : 'pointer'};
+  cursor: ${props => (props.mode === 'hover' ? 'default' : 'pointer')};
 `;
 
-const TargetSvgContainer = styled.g`
+const TargetSvgContainer = styled.g<{ mode: string }>`
   cursor: ${props => (props.mode === 'hover' ? 'default' : 'pointer')};
 `;
 
+interface WithTooltipPureProps {
+  svg?: boolean;
+  trigger?: 'none' | 'hover' | 'click' | 'right-click';
+  closeOnClick?: boolean;
+  placement?: Placement;
+  modifiers?: any;
```

Comment:
maybe use `unknown` instead of `any`?!  Here is a [great explanation around `any`, `never`, and `unknown`](https://blog.logrocket.com/when-to-use-never-and-unknown-in-typescript-5e4d6c5799ad) and when to use them. In general, I only found typing function with arbitrary numbers of arguments will use `any`.

Reply:
Fixed! Just as for `placement` I have used types from `popper.js` (the underlying library used for tooltips) so no more `any` here. I sometimes use `unknown` but it's not a reflex yet, soon ;)